### PR TITLE
Team02 Milestone Additional Features

### PR DESCRIPTION
**Pull request by timbittner**
See https://github.com/Jooshua/e4CompareFramework-team2/milestone/3 for details.

Features:
- Make Treeview DragSource and enhance Drag-and-Drop-Functions (Recursive-Copy-Options from filesystem)
- Replace clumsy Wizards with modern equivalents:
  - Replace Rename-Wizard with inline-editing
  - New Folder Quick-Action, 
  - Import File dialog
  - Added New File dialog
- Allow Multiselection in TreeView
- Toolbar with filter option & ability to change View-Representation

### CODE REVIEW

File: de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/CustomTreeCell.java [hunk 1]
```diff
@@ -1,53 +1,292 @@
...
+	/**
+	 * Will be invoked by pressing ESCAPE Key (default behavior by JavaFX)
+	 */
+	@Override
+	public void cancelEdit() {
+		super.cancelEdit();
+		setText(getItem().toString());
+		setGraphic(fileImageProvider.getImage(getItem()));
+	}
+
```
Comment (KamilRosiak) [inline de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/CustomTreeCell.java:199]: java doc

File: de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/controller/ProjectExplorerToolBarController.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	 * @return The created button
+	 */
+	private Button createToolbarButton(String tooltip, String imagePath, EventHandler<ActionEvent> value) {
+		ImageView image = services.imageService.getFXImage(null, imagePath);
+		Button button = new Button("", image);
+		button.setTooltip(new Tooltip(tooltip));
+		button.setOnAction(value);
+		bar.getItems().add(button);
+		return button;
+	}
```
Comment (KamilRosiak) [inline de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/controller/ProjectExplorerToolBarController.java:165]: Vielleicht in eine Factory/Builder Klasse refactoren guckt mal de.tu_bs.cs.isf.e4cf.core.gui.java_fx.util.JavaFXBuilder

File: de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/handlers/FileImportHandler.java [hunk 3]
```diff
@@ -1,92 +1,59 @@
...
-		}
-	}
-
-	private FileImportWizard buildFileImportWizard(Shell shell, RCPImageService imageService) {
-		ImageDescriptor image = imageService.getImageDescriptor(null, FILE_ICON_PATH);
-		FileImportWizard wizard = new FileImportWizard(shell, image);
-		return wizard;
-	}
-
 	@CanExecute
```
Comment (KamilRosiak) [inline de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/handlers/FileImportHandler.java:54]: Wo ist die methode hin mit der @CanExecute ?

File: de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/wizards/base_wizard/Wizard.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.layout.StackPane;
+
+/**
+ * Basic wizard infrastructure class,
+ * provides logic for navigating between pages
+ * 
+ */
+public class Wizard extends StackPane {
```
Comment (KamilRosiak) [inline de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/wizards/base_wizard/Wizard.java:14]: Java Doc für die Methoden pls

**Review by KamilRosiak — COMMENTED** [review]
Good Job :) only JavaDoc is missing.

**Comment by Jooshua** [comment]
We will resolve the comments in the PR on the main repo (https://github.com/TUBS-ISF/e4CompareFramework/pull/8). So no review from team03 is required anymore. @Hassan-smaoui can you please close this PR? Thanks :)

**Pull request closed without merging**